### PR TITLE
fix(seahorn): call assume with right args

### PR DESCRIPTION
**Pull request by priyasiddharth**

### CODE REVIEW

**Comment by priyasiddharth** [comment]
@googlebot I signed it!

**Comment by priyasiddharth** [comment]
@fshaked can you PTAL? Thanks

**Comment by fshaked** [comment]
Thanks for submitting.

Most of the changes are not relevant, and will be fixed by another pull request @alastairreid is working on.

Do we still need the `if` around the call to `__VERIFIER_assume`? If we keep the `if` does the fix really change anything?

File: verification-annotations/src/seahorn.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ pub fn abort() -> ! {
 /// Any paths found must satisfy this assumption.
 pub fn assume(pred: bool) {
     if ! pred {
-        unsafe { __VERIFIER_assume(0); }
+        unsafe { __VERIFIER_assume(pred as i32); }
```
Comment (alastairreid) [inline verification-annotations/src/seahorn.rs:41]: Is there an efficiency difference in SeaHorn between

"if !pred { unsafe { __VERIFIER_assume(0)} }"

and 

"__VERIFIER_assume(pred as i32)"

I think they mean the same but the latter maybe creates less branches/merges?

File: cargo-verify/src/main.rs [hunk 2]
```diff
@@ -429,7 +429,7 @@ fn build(opt: &Opt, package: &str, target: &str) -> CVResult<PathBuf> {
     Command::new("llvm-link")
         .arg("-o")
         .arg(&new_bc_file)
-        .arg(runtime)
+        //.arg(runtime)
```
Comment (alastairreid) [inline cargo-verify/src/main.rs:432]: I have just merged a fix for this problem into the main branch.
After starting docker, run 'docker/init' which will build the runtime for both KLEE and SeaHorn as well as building the cargo-verify and the rvt-patch-llvm tools.

**Review by alastairreid — COMMENTED** [review]
Thanks for sharing this with us - I think we have now fixed these issues.

**Comment by alastairreid** [comment]
I think we have fixed all these - please reopen if we missed something.

File: verification-annotations/src/seahorn.rs (hunk 1 above)
Comment (priyasiddharth) [reply verification-annotations/src/seahorn.rs:41]: Sorry I missed reading the `if`. This is not needed. 

thanks

**Pull request closed without merging**